fdbserver/TagPartitionedLogSystem.actor.cpp

```diff
@@ -2431,7 +2434,8 @@ ACTOR Future<Void> TagPartitionedLogSystem::epochEnd(Reference<AsyncVar<Referenc
 			lockResult.epochEnd = old.epochEnd;
 			lockResult.logSet = old.tLogs[0];
 			for (int t = 0; t < old.tLogs[0]->logServers.size(); t++) {
-				lockResult.replies.push_back(TagPartitionedLogSystem::lockTLog(dbgid, old.tLogs[0]->logServers[t]));
+				lockResult.replies.push_back(
+				    TagPartitionedLogSystem::lockTLog(dbgid, old.tLogs[0]->logServers[t], lockResultsInterf));
```

Comment:
If I read correctly, the intention is to update `lockResultsInterf` in the `lockTLog()` call. However, this parameter is passed in as `Optional<std::map<>>` type, so the map is copy-constructed inside `lockTLog()`. As a result, after the call completes, `lockResultsInterf` will not be updated.

Reply:
thanks for catching that.

Reply:
It appears this map is unsafe on the stack. The actor cannot access it if the callers stack was dropped. I will store the map in a Reference.